[BUG] error: no match for call to '(const eve::detail::callable_object<eve::tag::is_even_>) (long unsigned int)'

**Describe the bug**
```
In file included from /usr/local/include/eve/module/math/regular/geommean.hpp:70,
                 from /usr/local/include/eve/module/math/regular/math.hpp:59,
                 from /usr/local/include/eve/module/math.hpp:107,
                 from TEST.cpp:1:
/usr/local/include/eve/module/math/regular/impl/geommean.hpp: In function 'eve::common_value_t<T0, T1, Ts ...> eve::detail::geommean_(const eve::detail::delay_t&, const eve::cpu_&, T0, T1, Ts ...)':
/usr/local/include/eve/module/math/regular/impl/geommean.hpp:67:25: error: no match for call to '(const eve::detail::callable_object<eve::tag::is_even_>) (long unsigned int)'
   67 |   return if_else(is_even(sizeof...(args)) && is_ltz(sgn), eve::allbits, sgn * that);
      |                  ~~~~~~~^~~~~~~~~~~~~~~~~
In file included from /usr/local/include/eve/detail/function/bit_cast.hpp:11,
                 from /usr/local/include/eve/module/core/constant/allbits.hpp:11,
                 from /usr/local/include/eve/module/core/constant/core.hpp:10,
                 from /usr/local/include/eve/module/core.hpp:94,
                 from /usr/local/include/eve/module/math/constant/catalan.hpp:10,
                 from /usr/local/include/eve/module/math/constant/math.hpp:10,
                 from /usr/local/include/eve/module/math.hpp:105,
                 from TEST.cpp:1:
/usr/local/include/eve/detail/overload.hpp:79:48: note: candidate: 'template<class ... Args>  requires (tag_dispatchable<eve::tag::is_even_, Args ...>) || (supports_is_even_<Args ...>) constexpr decltype(auto) eve::detail::callable_object<eve::tag::is_even_, Dummy>::operator()(Args&& ...) const [with Args = {Args ...}; Dummy = void]'
   79 |       EVE_FORCEINLINE constexpr decltype(auto) operator()(Args &&... args) const noexcept          \
      |                                                ^~~~~~~~
/usr/local/include/eve/detail/overload.hpp:119:3: note: in expansion of macro 'EVE_DECLARE_CALLABLE'
  119 |   EVE_DECLARE_CALLABLE(TAG,NAME)                                                                    \
      |   ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/eve/detail/overload.hpp:129:3: note: in expansion of macro 'EVE_IMPLEMENT_CALLABLE'
  129 |   EVE_IMPLEMENT_CALLABLE(TAG,NAME)                                                                  \
      |   ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/eve/module/core/regular/is_even.hpp:64:1: note: in expansion of macro 'EVE_MAKE_CALLABLE'
   64 | EVE_MAKE_CALLABLE(is_even_, is_even);
      | ^~~~~~~~~~~~~~~~~
/usr/local/include/eve/detail/overload.hpp:79:48: note:   template argument deduction/substitution failed:
   79 |       EVE_FORCEINLINE constexpr decltype(auto) operator()(Args &&... args) const noexcept          \
      |                                                ^~~~~~~~
/usr/local/include/eve/detail/overload.hpp:119:3: note: in expansion of macro 'EVE_DECLARE_CALLABLE'
  119 |   EVE_DECLARE_CALLABLE(TAG,NAME)                                                                    \
      |   ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/eve/detail/overload.hpp:129:3: note: in expansion of macro 'EVE_IMPLEMENT_CALLABLE'
  129 |   EVE_IMPLEMENT_CALLABLE(TAG,NAME)                                                                  \
      |   ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/eve/module/core/regular/is_even.hpp:64:1: note: in expansion of macro 'EVE_MAKE_CALLABLE'
   64 | EVE_MAKE_CALLABLE(is_even_, is_even);
      | ^~~~~~~~~~~~~~~~~
/usr/local/include/eve/detail/overload.hpp:79:48: note: constraints not satisfied
   79 |       EVE_FORCEINLINE constexpr decltype(auto) operator()(Args &&... args) const noexcept          \
      |                                                ^~~~~~~~
/usr/local/include/eve/detail/overload.hpp:119:3: note: in expansion of macro 'EVE_DECLARE_CALLABLE'
  119 |   EVE_DECLARE_CALLABLE(TAG,NAME)                                                                    \
      |   ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/eve/detail/overload.hpp:129:3: note: in expansion of macro 'EVE_IMPLEMENT_CALLABLE'
  129 |   EVE_IMPLEMENT_CALLABLE(TAG,NAME)                                                                  \
      |   ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/eve/module/core/regular/is_even.hpp:64:1: note: in expansion of macro 'EVE_MAKE_CALLABLE'
   64 | EVE_MAKE_CALLABLE(is_even_, is_even);
      | ^~~~~~~~~~~~~~~~~
/usr/local/include/eve/module/math/regular/impl/geommean.hpp: In substitution of 'template<class ... Args>  requires (tag_dispatchable<eve::tag::is_even_, Args ...>) || (supports_is_even_<Args ...>) constexpr decltype(auto) eve::detail::callable_object<eve::tag::is_even_>::operator()<Args ...>(Args&& ...) const [with Args = void]':
/usr/local/include/eve/module/math/regular/impl/geommean.hpp:67:25:   required from here
/usr/local/include/eve/module/core/regular/is_even.hpp:64:1:   required by the constraints of 'template<class Dummy> template<class ... Args>  requires (tag_dispatchable<eve::tag::is_even_, Args ...>) || (supports_is_even_<Args ...>) constexpr decltype(auto) eve::detail::callable_object<eve::tag::is_even_, Dummy>::operator()(Args&& ...) const'
/usr/local/include/eve/detail/overload.hpp:77:17: note: no operand of the disjunction is satisfied
   76 |       requires  (   tag_dispatchable<tag_type,Args...>                                             \
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   77 |                 ||  supports_ ## TAG<Args...>                                                      \
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   78 |                 )                                                                                  \
      |                 ~
/usr/local/include/eve/detail/overload.hpp:119:3: note: in expansion of macro 'EVE_DECLARE_CALLABLE'
  119 |   EVE_DECLARE_CALLABLE(TAG,NAME)                                                                    \
      |   ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/eve/detail/overload.hpp:129:3: note: in expansion of macro 'EVE_IMPLEMENT_CALLABLE'
  129 |   EVE_IMPLEMENT_CALLABLE(TAG,NAME)                                                                  \
      |   ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/eve/module/core/regular/is_even.hpp:64:1: note: in expansion of macro 'EVE_MAKE_CALLABLE'
   64 | EVE_MAKE_CALLABLE(is_even_, is_even);
      | ^~~~~~~~~~~~~~~~~
cc1plus: note: set '-fconcepts-diagnostics-depth=' to at least 2 for more detail
```

**To Reproduce**
The command is `g++ -O2 -std=c++2a  TEST.cpp`

```
#include "eve/module/math.hpp"
#include "eve/wide.hpp"

float a = 2;
float x[16] = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16};
float y[16] = {17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32};
float res[16] = {0};

void axpy(float a, float x[16], float y[16], float res[16]){
        
    std::size_t vec_size = 16 - 16 % 4;
    
    for(long int i = 0; i < vec_size; i += 4){
        eve::wide<float> x_simd(&x[i]),y_simd(&y[i]),res_simd;
        res_simd = eve::fma(a,x_simd,y_simd);
        eve::store(res_simd,&res[i]);
    }

    for(std::size_t i = vec_size; i < 16; ++i)
    {
        res[i] = a * x[i] + y[i];
    }
}

int main()
{
    axpy(a,x,y,res);
}
```


**Expected behavior**
No error occurs.

**Setup:**
 - Compiler and compiler's version: gcc version 11.3.0 (Homebrew GCC 11.3.0) 
 - OS: aarch64-apple-darwin21
 - Processor Architecture AARCH64
 - Instructions Set used: ARM

--- a/include/eve/module/math/regular/impl/geommean.hpp
+++ b/include/eve/module/math/regular/impl/geommean.hpp
@@ -35,7 +35,7 @@ EVE_FORCEINLINE auto
 geommean_(EVE_SUPPORTS(cpu_),
           C const& cond,
           U const& t,
-          V const& f) noexcept 
+          V const& f) noexcept
 -> decltype( if_else(cond, geommean(t, f), t) )
 {
   return mask_op(cond, eve::geommean, t, f);
@@ -64,9 +64,10 @@ geommean_(EVE_SUPPORTS(cpu_),
     return avg * pow_abs(r_t(x), invn);
   };
   ((that = next(that, args)), ...);
-  return if_else(is_even(sizeof...(args)) && is_ltz(sgn), eve::allbits, sgn * that);
-}
 
+  constexpr std::uint64_t sz = sizeof...(args);
+  return if_else(eve::is_even(sz) && is_ltz(sgn), eve::allbits, sgn * that);
+}
 
 //================================================================================================
 // tuples

